packages/cloudflare/src/utils/wasm-module-loader.ts

```diff
@@ -77,51 +97,105 @@ export function wasmModuleLoader({
 			const chunkId = this.emitFile({
 				type: 'prebuilt-chunk',
 				fileName: `${assetName}.mjs`,
-				code: base64Module,
+				code: inlineModule,
 			});
 
-			return `import wasmModule from "__WASM_ASSET__${chunkId}.wasm.mjs";export default wasmModule;`;
+			return `import module from "${MAGIC_STRING}${chunkId}.${importAdapter.extension}.mjs";export default module;`;
 		},
 
-		// output original wasm file relative to the chunk
+		// output original wasm file relative to the chunk now that chunking has been achieved
 		renderChunk(code, chunk, _) {
 			if (isDev) return;
 
-			if (!/__WASM_ASSET__/g.test(code)) return;
-
-			const isPrerendered = Object.keys(chunk.modules).some(
-				(moduleId) => this.getModuleInfo(moduleId)?.meta?.astro?.pageOptions?.prerender === true
-			);
-
-			let final = code;
-
-			// SSR
-			if (!isPrerendered) {
-				final = code.replaceAll(/__WASM_ASSET__([A-Za-z\d]+).wasm.mjs/g, (s, assetId) => {
-					const fileName = this.getFileName(assetId).replace(/\.mjs$/, '');
-					const relativePath = path
-						.relative(path.dirname(chunk.fileName), fileName)
-						.replaceAll('\\', '/'); // fix windows paths for import
-					return `./${relativePath}`;
-				});
+			if (!code.includes(MAGIC_STRING)) return;
+
+			// SSR will need the .mjs suffix removed from the import before this works in cloudflare, but this is done as a final step
+			// so as to support prerendering from nodejs runtime
+			let replaced = code;
+			console.log(chunk.fileName);
+			for (const loader of enabledAdapters) {
+				replaced = replaced.replaceAll(
+					new RegExp(`${MAGIC_STRING}([A-Za-z\\d]+)\\.${loader.extension}\\.mjs`, 'g'),
+					(s, assetId) => {
+						const fileName = this.getFileName(assetId);
+						const relativePath = path
+							.relative(path.dirname(chunk.fileName), fileName)
+							.replaceAll('\\', '/'); // fix windows paths for import
+						console.log('for assetId', assetId, 'fileName', fileName, 'relativePath', relativePath);
+						replacements.push({
+							fileName,
+							cloudflareImport: relativePath.replace(/\.mjs$/, ''),
+							nodejsImport: relativePath,
+						});
+						return `./${relativePath}`;
+					}
+				);
 			}
-
-			// SSG
-			if (isPrerendered) {
-				final = code.replaceAll(/__WASM_ASSET__([A-Za-z\d]+).wasm.mjs/g, (s, assetId) => {
-					const fileName = this.getFileName(assetId);
-					const relativePath = path
-						.relative(path.dirname(chunk.fileName), fileName)
-						.replaceAll('\\', '/'); // fix windows paths for import
-					return `./${relativePath}`;
-				});
+			if (replaced.includes(MAGIC_STRING)) {
+				console.error('failed to replace', replaced);
 			}
 
-			return { code: final };
+			return { code: replaced };
+		},
+
+		/**
+		 * Once prerendering is complete, restore the imports in the _worker.js to cloudflare compatible ones, removing the .mjs suffix
+		 */
+		async afterBuildCompleted(config: AstroConfig) {
```

Comment:
this is the ugliest part of the change. A quick final string replace file fixup so that the import path is what's needed for cloudflare. Its pretty well targeted based on the prebuilt replacements array, so its not a super messy string replace